src/PointwiseFunctions/GeneralRelativity/KerrSchildCoords.cpp

```diff
@@ -0,0 +1,111 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/GeneralRelativity/KerrSchildCoords.hpp"
+
+#include <cmath>
+#include <cstddef>
+#include <pup.h>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+/// \cond
+namespace gr {
+
+KerrSchildCoords::KerrSchildCoords(const double spin_parameter) noexcept
+    : spin_parameter_(spin_parameter) {}
+
+void KerrSchildCoords::pup(PUP::er& p) noexcept { p | spin_parameter_; }
+
+template <typename DataType>
+Scalar<DataType> KerrSchildCoords::r_coord_squared(
+    const tnsr::I<DataType, 3, Frame::Inertial>& cartesian_coords) const
+    noexcept {
+  const double a_squared = square(spin_parameter_);
+  const DataType temp =
+      0.5 * (square(cartesian_coords.get(0)) + square(cartesian_coords.get(1)) +
+             square(cartesian_coords.get(2)) - a_squared);
+  return Scalar<DataType>{
+      temp + sqrt(temp * temp + a_squared * square(cartesian_coords.get(2)))};
+}
+
+template <typename DataType>
+tnsr::Ij<DataType, 3, Frame::Inertial> KerrSchildCoords::jacobian_matrix(
+    const tnsr::I<DataType, 3, Frame::Inertial>& cartesian_coords) const
+    noexcept {
+  auto result = make_with_value<tnsr::Ij<DataType, 3, Frame::Inertial>>(
+      cartesian_coords, 0.0);
+
+  const DataType x = cartesian_coords.get(0);
+  const DataType y = cartesian_coords.get(1);
+  const DataType z = cartesian_coords.get(2);
+  const DataType r_squared =
+      get(KerrSchildCoords::r_coord_squared(cartesian_coords));
+  const DataType r = sqrt(r_squared);
+  const double a_squared = square(spin_parameter_);
+
+  DataType r_squared_plus_a_squared = r_squared + a_squared;
+  const DataType prefactor_dth =
+      z * sqrt(r_squared_plus_a_squared / (square(x) + square(y))) / r;
+  const DataType prefactor_dr = 1.0 / std::move(r_squared_plus_a_squared);
+
+  result.get(0, 0) = prefactor_dr * (x * r - spin_parameter_ * y);
```

Comment:
Since you know the indices at compile time, should you use get<0,0>(result) instead here and following?